src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -132,7 +148,21 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                     return;
                 }
 
-                bindingContext.Result = ModelBindingResult.Success(model);
+                if (_isBindingRequired && model == null)
```

Comment:
I'm of the opinion that we should kill Line 108 with fire. Lou felt pretty strongly about this as well and look at what happened to him 😲 

Formatters should decide for themselves how 'no input' is defined. This sounds like a change to `InputFormatterResult.